app/controllers/citizens/student_finances_controller.rb

```diff
@@ -1,27 +1,38 @@
 module Citizens
   class StudentFinancesController < CitizenBaseController
     def show
-      @form = LegalAidApplications::StudentFinanceForm.new(model: legal_aid_application)
+      @form = ::StudentFinances::AnnualAmountForm.new(model: irregular_income)
     end
 
     def update
-      @form = LegalAidApplications::StudentFinanceForm.new(form_params)
-
-      if @form.save
-        go_forward
+      @form = ::StudentFinances::AnnualAmountForm.new(form_params)
+      if student_finance
+        legal_aid_application.update!(student_finance:)
+        if student_finance == "false" || @form.save
+          return go_forward
+        end
```

Comment:
```suggestion
        legal_aid_application.update!(student_finance:) if student_finance
        return go_forward if student_finance == "false" || @form.save
```

Reply:
@chris-groves I don't think the suggested change will work, because the if statement logic needs the else clause - unfortunately the update action has become quite complex because of trying to combine the student_finances and annual_amounts update controller actions 🤔